app/src/RequestMoneyValidator/Dto/TransactionDto.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace App\src\RequestMoneyValidator\Dto;
+
+use App\Enum\Currency;
+
+readonly final class TransactionDto
+{
+    private function __construct(
+        public float $amount,
+        public Currency $currency
+    ) {
+    }
+
+    public static function fromRequest(array $data): self
+    {
+        return new self(
+            amount: round($data['amount'], 2),
```

Comment:
same problems as for RequestDto - the `float` is not the best way to operate, this point was raised in a task description. 

This approach is too bad and is used in some apps, though it is a bit outdated imo.

Instead of converting to cents and then back to currency (which is not only painful but aslo unsafe) – the better approach is to cast the money to `string` and use `bcmath` php lib for all the math operations and compares.